### ISSUE TITLE
Incorrect IOU definition

### ISSUE THREAD

**Issue by nzw0301** [4uiiurz1/pytorch-nested-unet#36]
[The current IOU implementation](https://github.com/4uiiurz1/pytorch-nested-unet/blob/master/metrics.py#L15-L18) has calculated the union and intersection over samples in a mini-batch, then it has calculated the IOU score. But I think IOU should be calculated per sample then take the mean over the samples in the mini-batch. 

I think the following code is correct when the number of classes is 1:

```python
intersection = (output_ & target_).sum(axis=(1, 2, 3))
union = (output_ | target_).sum(axis=(1, 2, 3))

return (intersection + smooth) / (union + smooth).mean()
```

**Comment by zhou-rui1** [comment]
Hi, will that IOU score higher?

**Comment by nzw0301** [comment]
@zhou-rui1 I don't think so. The current implementation might output a much higher score.
